packages/docusaurus-theme-redoc/src/types/common.ts

```diff
@@ -1,17 +1,45 @@
 import type { Props as LayoutProps } from '@theme/Layout';
-import type { ObjectDescriptionProps } from 'redoc';
+import type { ObjectDescriptionProps, RedocRawOptions } from 'redoc';
 import type { OpenAPISpec } from 'redoc/typings/types';
 
 export type ParsedSpec = OpenAPISpec;
 
 export interface SpecProps {
+  /**
+   * Spec to use, already loaded previously
+   */
   spec: ParsedSpec;
-  url?: string;
-  isSpecFile?: boolean;
+  /**
+   * When spec not provided, load the spec from docusaurus config
+   * fallback to first configuration if not provided
+   */
+  id?: string;
```

Comment:
Isn't id used only in Mdx? This will add it to `Redoc` component also

Reply:
SpecProps is now the type of the `useSpec` function.

So the Mdx component is not impacted